contributing-docs/11_documentation_building.rst

```diff
@@ -0,0 +1,160 @@
+   installed and usually the library is in the ``lib`` subfolder of it. An example of line added
+   to one of the ``.rc`` files:
+
+   .. code-block:: bash
+
+      export PYENCHANT_LIBRARY_PATH=/opt/homebrew/Cellar/enchant/2.8.2/lib/libenchant-2.dylib
+
+   Also for some of the providers you might need to have trouble in installing dependencies. The ``uv run``
+   command will automatically install the needed dependencies for you but some of them might need extra
+   system dependencies. You can always revert to ``breeze`` way of building the documentation in this
+   case (see below) in case you have problems with ``uv run`` command not being able to install
+   the dependencies.
+
+Once you have enchant installed and configured, you can build the documentation very easily.
+
+
+
+First thing to do, is to sync the inter-sphinx inventory. You can do it only once and refresh it periodically
+when you find that your documentation cannot be build because of missing inter-sphinx inventory or links
+The inventory is basically a list of links to other documentation that is used in the Airflow documentation
+and our build issues warnings if the inventory links are missing.
+
+.. code-block:: bash
+
+    cd devel-common
+    uv run src/sphinx_exts/docs_build/fetch_inventories.py
+
+When you want to refresh the inventory, you can run the same command with ``clean`` flag.
+
+.. code-block:: bash
+
+    cd devel-common
+    uv run src/sphinx_exts/docs_build/fetch_inventories.py clean
+
+
+To run docs building go to the directory where documentation is placed (usually ``doc`` subfolder of
+a distribution you want to build the documentation for) and run, but some distributions do not have ``docs``
+subfolder if they are ``doc-only``.
+
+.. code-block:: bash
+      cd <YOUR-DISTRIBUTION>/docs
+      uv run --group doc sphinx-build -T --color -b html  . _build
+
+To check spelling:
+
+.. code-block:: bash
+      cd <YOUR-DISTRIBUTION>/docs
+      uv run --group doc sphinx-build -T --color -b spelling  . _build
+
+But most importantly you can run a documentation server that will expose the built documentation and
+enable auto-refreshing build (automatically refreshing the browser when you change the documentation files):
+
+.. code-block:: bash
+      cd <YOUR-DISTRIBUTION>/docs
+      uv run sphinx-autobuild -T --color -b html  . _build
+
+That should allow you to build the documentation and see the changes in your browser.
+
+Note that those build commands are not the same as the "breeze build-docs" command. The ``breeze build-docs``
+command is a more complete way of building the whole documentation together for all distributions.
+
+.. warning::
+
+   Sometimes - when you have references between distributions, you might need to build those distributions
+   first and then build the documentation for the distribution that depends on them, but in this case you will
```

Comment:
What if there are dependencies in both directions? Do you get stuck with a "circular import"-like issue?

Reply:
Generally our current logic in `breeze build-docs` is that we assume there are no circular "changes".  There might be circular dependencies - that's fine and it is handled by `inventories` - we publish latest inventories in main and when you run `fetch_inventories.py clean` you will get the latest version. Also I **think ** what happens there is that even if the new build fails, it will produce enough of generated files, that "references" to the new document will work. 

I am also going to improve that part a bit and see if we can easily unify where `_build` files are generated.

Reply:
I addressed it by simply repurposing our `build_docs.py` to be run locally - an d it should handle all the same complexity as `breeze build-docs` - including solving circular references.